src/graphql/node/control.rs

```diff
@@ -74,14 +91,924 @@ impl NodeControlMutation {
+            updated_setting.review_port = settings_draft.review_port;
+            updated_setting.review_web_port = settings_draft.review_web_port;
+        }
+
+        if update_module_specific_settings.hog {
+            updated_setting.hog = settings_draft.hog;
+            updated_setting.hog_review_ip = settings_draft.hog_review_ip;
+            updated_setting.hog_review_port = settings_draft.hog_review_port;
+            updated_setting.hog_giganto_ip = settings_draft.hog_giganto_ip;
+            updated_setting.hog_giganto_port = settings_draft.hog_giganto_port;
+            updated_setting.protocols = settings_draft.protocols;
+            updated_setting.protocol_list = settings_draft.protocol_list.clone();
+            updated_setting.sensors = settings_draft.sensors;
+            updated_setting.sensor_list = settings_draft.sensor_list.clone();
+        }
+
+        if update_module_specific_settings.reconverge {
+            updated_setting.reconverge = settings_draft.reconverge;
+            updated_setting.reconverge_review_ip = settings_draft.reconverge_review_ip;
+            updated_setting.reconverge_review_port = settings_draft.reconverge_review_port;
+            updated_setting.reconverge_giganto_ip = settings_draft.reconverge_giganto_ip;
+            updated_setting.reconverge_giganto_port = settings_draft.reconverge_giganto_port;
+        }
+
+        if update_module_specific_settings.piglet {
+            updated_setting.piglet = settings_draft.piglet;
+            updated_setting.piglet_review_ip = settings_draft.piglet_review_ip;
+            updated_setting.piglet_review_port = settings_draft.piglet_review_port;
+            updated_setting.piglet_giganto_ip = settings_draft.piglet_giganto_ip;
+            updated_setting.piglet_giganto_port = settings_draft.piglet_giganto_port;
+            updated_setting.save_packets = settings_draft.save_packets;
+            updated_setting.http = settings_draft.http;
+            updated_setting.office = settings_draft.office;
+            updated_setting.exe = settings_draft.exe;
+            updated_setting.pdf = settings_draft.pdf;
+            updated_setting.html = settings_draft.html;
+            updated_setting.txt = settings_draft.txt;
+            updated_setting.smtp_eml = settings_draft.smtp_eml;
+            updated_setting.ftp = settings_draft.ftp;
+        }
+    }
+
+    updated_node.settings = Some(updated_setting);
+}
+
+fn should_broadcast_customer_change(node: &Node) -> (bool, u32) {
+    match (node.settings.as_ref(), node.settings_draft.as_ref()) {
+        (None, Some(settings_draft)) => (settings_draft.review, settings_draft.customer_id),
+        (Some(settings), Some(settings_draft)) => (
+            settings_draft.review && settings_draft.customer_id != settings.customer_id,
+            settings_draft.customer_id,
+        ),
+        (_, None) => {
+            info!("When there is no settings draft, customer change should not be broadcasted.");
```

Comment:
Could you tell me why this message is necessary?

Reply:
done